### PR TITLE
Add net 4.5.2 and upgrade MSpec to v1

### PR DESCRIPTION
**Pull request by mikeblakeuk**
- [x] Add support for .net 4.5.2 (for users who are still on cloud services!)
- [x] Upgrade Machine.Specifications to v1
- [x] Include nuget on build for testing

### CODE REVIEW

File: src/Machine.Specifications.Runner.VisualStudio/Machine.Specifications.Runner.VisualStudio.csproj [hunk 1]
```diff
@@ -1,8 +1,8 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <Version>0.1.0</Version>
-    <TargetFrameworks>netcoreapp2.1;net46</TargetFrameworks>
+    <Version>0.2.0</Version>
```
Comment (robertcoltheart) [inline src/Machine.Specifications.Runner.VisualStudio/Machine.Specifications.Runner.VisualStudio.csproj]: No need to change this, github takes care of versioning

File: src/Machine.Specifications.Runner.VisualStudio/Machine.Specifications.Runner.VisualStudio.csproj [hunk 2]
```diff
@@ -22,19 +22,24 @@
 
   <ItemGroup>
     <PackageReference Include="Mono.Cecil" Version="[0.10-*, 0.11)" />
-    <PackageReference Include="Machine.Specifications" Version="[0.11.0, 1.0.0]" PrivateAssets="All" />
+    <PackageReference Include="Machine.Specifications" Version="1.0.0" PrivateAssets="All" />
```
Comment (robertcoltheart) [inline src/Machine.Specifications.Runner.VisualStudio/Machine.Specifications.Runner.VisualStudio.csproj]: The version range was deliberate, can you revert this?

File: src/Machine.Specifications.Runner.VisualStudio/Machine.Specifications.Runner.VisualStudio.csproj [hunk 3]
```diff
@@ -22,19 +22,24 @@
...
   </ItemGroup>
 
   <ItemGroup Condition="'$(TargetFramework)'=='netcoreapp2.1'">
     <PackageReference Include="Microsoft.TestPlatform.ObjectModel" Version="[15.0.0,17.0.0)" PrivateAssets="All" />
   </ItemGroup>
-  
+
```
Comment (robertcoltheart) [inline src/Machine.Specifications.Runner.VisualStudio/Machine.Specifications.Runner.VisualStudio.csproj]: unnecessary whitespace

File: src/Machine.Specifications.Runner.VisualStudio/Machine.Specifications.Runner.VisualStudio.csproj [hunk 4]
```diff
@@ -22,19 +22,24 @@
...
   <ItemGroup Condition="'$(TargetFramework)'=='net46'">
     <PackageReference Include="Microsoft.TestPlatform.ObjectModel" Version="[11.0.0,17.0.0)" PrivateAssets="All" />
   </ItemGroup>
 
+  <ItemGroup Condition="'$(TargetFramework)'=='net452'">
+    <PackageReference Include="Microsoft.TestPlatform.ObjectModel" Version="[11.0,12.0]" PrivateAssets="All" />
```
Comment (robertcoltheart) [inline src/Machine.Specifications.Runner.VisualStudio/Machine.Specifications.Runner.VisualStudio.csproj]: Version range should be the same as net46. As above, lets just use net452 instead of net46.

File: src/Machine.Specifications.Runner.VisualStudio/Machine.Specifications.Runner.VisualStudio.csproj [hunk 5]
```diff
@@ -22,19 +22,24 @@
...
+  </ItemGroup>
+
   <PropertyGroup>
     <TargetsForTfmSpecificContentInPackage>$(TargetsForTfmSpecificContentInPackage);NetCorePackageItems;NetFrameworkPackageItems</TargetsForTfmSpecificContentInPackage>
+    <GeneratePackageOnBuild>true</GeneratePackageOnBuild>
```
Comment (robertcoltheart) [inline src/Machine.Specifications.Runner.VisualStudio/Machine.Specifications.Runner.VisualStudio.csproj]: This is annoying for local development, revert.

File: src/Machine.Specifications.Runner.VisualStudio/Machine.Specifications.Runner.VisualStudio.csproj [hunk 6]
```diff
@@ -1,8 +1,8 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <Version>0.1.0</Version>
-    <TargetFrameworks>netcoreapp2.1;net46</TargetFrameworks>
+    <Version>0.2.0</Version>
+    <TargetFrameworks>netcoreapp2.1;net46;net452</TargetFrameworks>
```
Comment (robertcoltheart) [inline src/Machine.Specifications.Runner.VisualStudio/Machine.Specifications.Runner.VisualStudio.csproj]: Let's just replace net46 with net452, no need to have both in the nuget.

File: src/Machine.Specifications.Runner.VisualStudio/Machine.Specifications.Runner.VisualStudio.csproj (hunk 4 above)
Comment (mikeblakeuk) [reply src/Machine.Specifications.Runner.VisualStudio/Machine.Specifications.Runner.VisualStudio.csproj]: The reason i did this is it looks like net46 CAN have a later version

**Comment by mikeblakeuk** [comment]
@robertcoltheart updated, let me know your thoughts. thanks

File: src/Machine.Specifications.Runner.VisualStudio/Machine.Specifications.Runner.VisualStudio.csproj (hunk 4 above)
Comment (robertcoltheart) [reply src/Machine.Specifications.Runner.VisualStudio/Machine.Specifications.Runner.VisualStudio.csproj]: Ah I see now, yep this should be fine.

**Comment by robertcoltheart** [comment]
Thanks for your contribution!

**Comment by robertcoltheart** [comment]
@mikeblakeuk Just out of curiosity, what's the use case for needing to support `net452`? Testing frameworks like xUnit are looking at base-lining to `netstandard2.0`, which means minimum versions of `netcoreapp2.1` and `net472`.

**Comment by mikeblakeuk** [comment]
We have code that runs on Azure Cloud Services that are pinned to net452 (sure to windows 2013 booting faster!). 
We are also looking to upgrade to 4.72 (or even netcore3) but not ready yet.
The main reason was to allow non R# users run tests locally in Visual Studio 2019.
It was also to see what the adapter was like in Azure DevOps pipelines (But might be hard to use if filters are not supported)

**Comment by robertcoltheart** [comment]
Got it, thanks.

**Comment by mikeblakeuk** [comment]
We have now moved to net472, windows2019
